streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -762,16 +762,22 @@ abstract class DStream[T: ClassTag] (
     if (!isInitialized) {
       throw new SparkException(this + " has not been initialized")
     }
-    if (!(fromTime - zeroTime).isMultipleOf(slideDuration)) {
-      logWarning("fromTime (" + fromTime + ") is not a multiple of slideDuration ("
-        + slideDuration + ")")
+
+    val alignedToTime = (toTime - zeroTime).isMultipleOf(slideDuration) match {
```

Comment:
I think match syntax for booleans is needlessly complicated. You can just write a more straightforward `if`.